.github/workflows/testing.yml

```
@@ -0,0 +1,253 @@
+
+      - name: Upload benchmark results
+        uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
+        with:
+          name: benchmark-results
+          path: benchmark-report.json
+
+  e2e-tests:
+    runs-on: ubuntu-latest
+    needs: [integration-tests]
+    permissions:
+      contents: read # Required for actions/checkout
```

YAMLlint (1.37.1)

[warning] too few spaces before comment: expected 2

(comments)